ulib/LowStar.RVector.fst

```diff
@@ -757,11 +757,14 @@ val alloc_reserve:
   HST.ST (rvector rg)
     (requires (fun h0 -> true))
     (ensures (fun h0 rv h1 ->
-      modifies (V.loc_vector rv) h0 h1 /\
+      modifies loc_none h0 h1 /\
       rv_inv h1 rv /\
       V.frameOf rv = rid /\
       V.size_of rv = 0ul /\
-      S.equal (as_seq h1 rv) S.empty))
+      S.equal (as_seq h1 rv) S.empty /\
+      Set.equal (Map.domain (HS.get_hmap h0))
```

Comment:
this post-condition is unusual, did you mean to use the `Stack` effect?

Reply:
I don't understand the difference - I just need it to be the same as on `LowStar.Vector.alloc_reserve`. Would some of this be implicit with the `Stack` effect?

Reply:
(Yes; will simplify both.)

Reply:
Actually, no, `Stack` requires `same_refs_in_all_regions m0 m1`, which isn't true here. Should it be true or is there a different effect that doesn't require that?